File: src/main/webapp/resources/core/js/services/produto.service.js
app.factory('ProdutoService', ['$http', function($http){
	
	var _save = function(produto){
		return $http.post(SERVER_APP + '/produto/', produto);
	}
	
	var _delete = function(id){
		return $http.get(SERVER_APP + '/produto/delete/' + id);
	}
	
	var _findAll = function(){
		return $http.get(SERVER_APP + '/produto/findall');
	}
	
	return {
		
		save: _save,
		
		delete: _delete,
		
		findAll: _findAll
		
	}
}])

File: src/main/java/com/stockmanager/services/ProdutoService.java
package com.stockmanager.services;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.stockmanager.entities.Produto;
import com.stockmanager.repositories.ProdutoRepository;

@Service
public class ProdutoService {

	@Autowired
	private ProdutoRepository produtoRepository;
	
	public Produto save(Produto produto){
		Produto saved = this.produtoRepository.save(produto);
		return saved;
	}
	
	public void delete(Long id){
		this.produtoRepository.delete(id);
	}

	public List<Produto> findAll() {
		return (List<Produto>) this.produtoRepository.findAll();
	}
}
